### DISCUSSION TITLE
Disclosure with button rendered as link?

### DISCUSSION THREAD

**Discussion by andioneto** [twilio-labs/paste#521] [General] [answered]
Hi folks, this is a follow up question/discussion to my question about [how to style the disclosure button in the primitive](https://github.com/twilio-labs/paste/discussions/517).

In my case, I am using the disclosure to conditionally show form sections to edit and additional features.
In a feedback session with the PM, they asked for this render to decrease noise on the page from all of the form sections.
I tried it out with a secondary button, but felt like it created too much noise on the page.
Here is a [sandbox of a demo form using the disclosure primitive](https://github.com/twilio-labs/paste/discussions/517) with the button rendered as a link in two different positions.
I think having the button rendered as a link _under_ the paragraph makes most sense, but I'm curious what you all think.
Also, is my use of the button rendered with link styling here appropriate?

**Comment by corinne-nagel** [comment, marked as answer]
I think having the button render under the paragraph makes the most sense!
I think the styling is correct, maybe reduce some of the padding on the left and right of the carrot by 2/4 pixels or so and move the carrot to the right of the text. Otherwise looks good!

**Reply by andioneto** [reply]
Awesome, thank you 😸

**Reply by andioneto** [reply]
@corinne-nagel below is the UI after adjusting the padding. FYI I had to create a styled `<span />` wrapper with `@emotion/styled` because the default padding for the button component is 6px and the disclosure rendered as a button do not allow for adjusting padding.
Please let me know if that is problematic or if you see anything below that still looks off. Thanks for the quick feedback 🙌 
<img width="1214" alt="Screen Shot 2020-06-22 at 3 43 26 PM" src="https://user-images.githubusercontent.com/18120906/85342337-36c68100-b49f-11ea-9e54-cca69a5ef0e9.png">

**Reply by aayushpi** [reply]
nice, did you use Box `as` span?

**Reply by andioneto** [reply]
No, I couldn't because `Box` doesn't accept any negative padding values. 😿 
The text span inside the button has padding that I couldn't modify. So instead I used `styled.span({ margin: '-.5em' })`.

**Reply by andioneto** [reply]
That's why the padding was indenting the button as link slightly off the alignment from the paragraph above.

**Comment by corinne-nagel** [comment]
Looks great to me! Thanks!
